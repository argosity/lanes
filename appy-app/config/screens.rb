
Lanes::Screen.define "base" do | screen |
    screen.description = "Base Screen for Appy App"
    screen.icon        = ""
    screen.group_id    = ""
    screen.model_class = ""
    screen.view_class  = "AppyApp.Screens.Base"
end
