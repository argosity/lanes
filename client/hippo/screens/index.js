import { observable, computed } from 'mobx';
import { map, filter, intersection } from 'lodash';
import Group from './group';
import Config from '../config';

const Screens = observable({

    all: observable.map(),

    get activeGroups() {
        const gids = Group.enabled_group_ids || map(Group.all, 'id');
        const groups = map(gids, gid => Group.forId(gid));
        return filter(groups, group => intersection(group.screens, this.active).length);
    },

    @computed get active() {
        return map(Config.screen_ids, id => this.all.get(id));
    },

    reset() {
        Config.screen_ids.clear();
    },

});

export default Screens;
