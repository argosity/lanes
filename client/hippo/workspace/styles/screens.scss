.screen {
    padding: 5px;
    width: 100%;

    > .first-run {
        font-size: 1.3rem;
        li + li {
            margin-top: 1rem;
        }
        .help {
            margin-top: 2rem;
            p { text-align: right; }
        }
    }
}
