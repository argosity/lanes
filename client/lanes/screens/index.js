import { observable, autorun } from 'mobx';
import { map, uniq } from 'lodash';
import Group from './group';
import Config from '../config';
import Sync from '../models/sync';

import user from '../user';

const Screens = observable({

    all: observable.map(),

    active: observable.array(),

    groups: observable.map(),

    get activeGroups() {
        return uniq(map(this.active, s => Group.forId(s.group_id)));
    },

    refresh() {
        Sync.perform(`${Config.api_path}/lanes/screens`).then(({ data: { screens: screenIds } }) => {
            this.active.replace(map(screenIds, id => this.all.get(id)));
        });
    },

});
export default Screens;

let previousLoggedIn = null;
autorun(() => {
    if (user.isLoggedIn !== previousLoggedIn){
        previousLoggedIn = user.isLoggedIn;
        Screens.refresh();
    }
});
