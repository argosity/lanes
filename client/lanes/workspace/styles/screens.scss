.screen {
    padding: 5px;
    //  overflow: hidden;
    width: 100%;
    .toolbar {
        margin-top: 10px;
    }

    .toolbar{
        @extend .navbar;
        @extend .navbar-default;

        .btn {
            @extend .btn-default;
            @extend .navbar-btn;
        }

        .arrow {
            border-style: solid;

            border-left-color: transparent;
            border-right-color: transparent;

            border-bottom-color: $popover-title-bg;
            border-top-color: $popover-title-bg !important; 

            border-width: 0 9px 9px;
            height: 0;
            left: 7px;
            margin-top: 10px;
            opacity: 1;
            position: absolute;
            top: -18px;
            transition: all 0.25s ease 0s;
            width: 0;
            z-index: 10;
        }
        .navbar-left {
            float: left;
        }
        .navbar-right{
            float: right;
            .arrow {
                left: unset;
                right: 10px;
            }
        }

        .dropdown-toggle {
            .badge{
                padding: 2px 2px 0;
                position: relative;
                font-size: 70%;
                top: -6px;
                right: 5px;
            }
            position: relative;
            padding-right: 2px;
            i {
                /* position: absolute; */
                /* left: 0px; */
            }
        }

    }

}
