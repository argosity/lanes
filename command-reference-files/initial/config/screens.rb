# An example of how to define a screen group
# Hippo::Screen.define_group 'appy-app-screens' do | group |
#     group.title       = "AppyApp Screens"
#     group.description = "Screens relating to AppyApp"
#     group.icon        = "heart"
# end

Hippo::Screen.for_extension 'appy-app' do | screens |

end
