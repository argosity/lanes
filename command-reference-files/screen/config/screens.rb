# An example of how to define a screen group
# Hippo::Screen.define_group 'appy-app-screens' do | group |
#     group.title       = "AppyApp Screens"
#     group.description = "Screens relating to AppyApp"
#     group.icon        = "heart"
# end

Hippo::Screen.for_extension 'appy-app' do | screens |
    screens.define "ready-set-go" do | screen |
        screen.title       = "Ready Set Go"
        screen.description = ""
        screen.icon        = ""
        screen.group_id    = ""
        screen.model_class = ""
        screen.view_class  = "ReadySetGo"
    end

end
