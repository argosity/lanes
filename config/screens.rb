Lanes::Screen.define_group "system-settings" do | group |
    group.title       = "Settings"
    group.description = "Modify system settings"
    group.icon        = "wrench"
end

Lanes::Screen.for_extension 'lanes' do | screens |

    # screens.define "user-preferences" do | screen |
    #     screen.title       = "User Preferences"
    #     screen.description = "Modify User Profile"
    #     screen.icon        = "user-secret"
    #     screen.group_id    = "system-settings"
    #     screen.model_class = "User"
    #     screen.view_class  = "UserPreferences"
    #     screen.asset  = 'user'
    # end
    screens.define "user-management" do | screen |
        screen.title       = "User Management"
        screen.description = "Add/Remove/Modify User Accounts"
        screen.icon        = "group"
        screen.group_id    = "system-settings"
        screen.model_class = "SystemSettings"
        screen.view_class  = "UserManagement"
        screen.asset       = 'lanes/access/screens/user-management'
    end

    screens.define "system-settings" do | screen |
        screen.title       = "System Settings"
        screen.description = "Update System Settings"
        screen.icon        = "cogs"
        screen.group_id    = "system-settings"
        screen.model_class = "SystemSettings"
        screen.asset       = 'lanes/screens/system-settings'
    end

    # screens.define "workspace-layout" do | screen |
    #     screen.url_prefix  = 'lanes/workspace'
    #     screen.js  = 'index.jsx'
    #     # screen.css = 'styles.scss'
    # end
end
