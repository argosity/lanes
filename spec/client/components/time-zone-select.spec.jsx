import React from 'react'; // eslint-disable-line no-unused-vars
import TZ from 'hippo/components/time-zone-select';
import { Snapshot } from 'hippo/testing/screens';


describe('TimeZone Component', () => {
    it('matches snapshot', () => {
        const oc = jest.fn();
        expect(Snapshot(<TZ onChange={oc} value="America/Chicago" />)).toMatchSnapshot();
    });
});
