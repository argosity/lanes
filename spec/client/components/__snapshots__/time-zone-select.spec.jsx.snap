// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeZone Component matches snapshot 1`] = `
.c6 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  fill: #666666;
  stroke: #666666;
  fill: #865CD6;
  stroke: #865CD6;
}

.c6 g {
  fill: inherit;
  stroke: inherit;
}

.c6 *:not([stroke])[fill="none"] {
  stroke-width: 0;
}

.c6 *[stroke*="#"],
.c6 *[STROKE*="#"] {
  stroke: inherit;
  fill: none;
}

.c6 *[fill-rule],
.c6 *[FILL-RULE],
.c6 *[fill*="#"],
.c6 *[FILL*="#"] {
  fill: inherit;
  stroke: none;
}

.c1 {
  box-sizing: border-box;
  cursor: pointer;
  outline: none;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font: inherit;
  margin: 0;
  background-color: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  border: none;
  padding: 0;
  text-align: inherit;
}

.c0 {
  font-family: 'Work Sans',Arial,sans-serif;
  font-size: 1em;
  line-height: 1.5;
  color: #444444;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: solid 1px rgba(0,0,0,0.33);
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 0;
  padding: 0;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  min-width: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  margin-left: 12px;
  margin-right: 12px;
  padding: 0;
}

.c4 {
  box-sizing: border-box;
  padding: 11px;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
  outline: none;
  background-color: transparent;
  color: inherit;
  font: inherit;
  font-weight: 700;
  margin: 0;
  width: 100%;
  border: none;
  width: 100%;
  -webkit-appearance: none;
}

.c4::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c4::-moz-placeholder {
  color: #AAAAAA;
}

.c4:-ms-input-placeholder {
  color: #AAAAAA;
}

.c4::-moz-focus-inner {
  border: none;
  outline: none;
}

.c3 {
  width: 100%;
}

@media only screen and (min-width:700px) {
  .c1 {
    -webkit-transition: 0.1s ease-in-out;
    transition: 0.1s ease-in-out;
  }
}

@media only screen and (max-width:699px) {
  .c2 {
    border: solid 1px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:699px) {
  .c2 {
    margin: 0;
  }
}

@media only screen and (max-width:699px) {
  .c2 {
    padding: 0;
  }
}

@media only screen and (max-width:699px) {
  .c5 {
    margin-left: 6px;
    margin-right: 6px;
  }
}

@media only screen and (max-width:699px) {
  .c5 {
    padding: 0;
  }
}

<div
  className="c0"
>
  <button
    aria-label="undefined, America/Chicago"
    className="c1"
    disabled={false}
    href={undefined}
    icon={undefined}
    id={undefined}
    label={undefined}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    type="button"
  >
    <div
      aria-hidden={true}
      aria-label={undefined}
      className="c2"
      direction="row"
    >
      <div
        className="c3"
      >
        <input
          autoComplete="off"
          className="c4"
          defaultValue={undefined}
          id={undefined}
          onFocus={[Function]}
          onInput={[Function]}
          onKeyDown={[Function]}
          placeholder={undefined}
          readOnly={true}
          size={undefined}
          style={
            Object {
              "cursor": "pointer",
            }
          }
          tabIndex="-1"
          type="text"
          value="America/Chicago"
        />
      </div>
      <div
        aria-label={undefined}
        className="c5"
        direction="column"
        style={
          Object {
            "minWidth": "auto",
          }
        }
      >
        <svg
          aria-label="FormDown"
          className="c6"
          color="brand"
          height="24px"
          role="img"
          size={undefined}
          version="1.1"
          viewBox="0 0 24 24"
          width="24px"
        >
          <polyline
            fill="none"
            points="18 9 12 15 6 9"
            stroke="#000"
            strokeWidth="2"
          />
        </svg>
      </div>
    </div>
  </button>
</div>
`;
